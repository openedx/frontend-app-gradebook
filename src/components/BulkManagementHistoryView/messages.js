import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  heading: {
    id: 'gradebook.BulkManagementHistoryView.heading',
    defaultMessage: 'Bulk Management History',
    description: 'Heading text for BulkManagement History Tab',
  },
  helpText: {
    id: 'gradebook.BulkManagementHistoryView',
    defaultMessage: 'Below is a log of previous grade imports.  To download a CSV of your gradebook and import grades for override, return to the Gradebook.  Please note, after importing grades, it may take a few seconds to process the override.',
    description: 'Bulk Management History View help text',
  },
  successDialog: {
    id: 'gradebook.BulkManagementHistoryView.successDialog',
    defaultMessage: 'CSV processing. File uploads may take several minutes to complete.',
    description: 'Success Dialog message in BulkManagement Tab File Upload Form',
  },
});

export default messages;
