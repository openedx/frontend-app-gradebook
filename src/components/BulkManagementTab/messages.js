import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  csvUploadLabel: {
    id: 'gradebook.BulkManagementTab.csvUploadLabel',
    defaultMessage: 'Upload Grade CSV',
    description: 'Button in BulkManagementTab Alerts',
  },
  heading: {
    id: 'gradebook.BulkManagementTab.heading',
    defaultMessage: 'Use this feature by downloading a CSV for bulk management, overriding grades locally, and coming back here to upload.',
    description: 'Heading text for BulkManagement Tab',
  },
  hint1: {
    id: 'gradebook.BulkManagementTab.hint1',
    defaultMessage: 'Results appear in the table below.',
    description: 'Hint text on BulkManagement Tab History Table',
  },
  hint2: {
    id: 'gradebook.BulkManagementTab.hint2',
    defaultMessage: 'Grade processing may take a few seconds.',
    description: 'Hint text on BulkManagement Tab History Table',
  },
  importBtnText: {
    id: 'gradebook.BulkManagementTab.importBtnText',
    defaultMessage: 'Import Grades',
    description: 'Button in BulkManagement Tab File Upload Form',
  },
  successDialog: {
    id: 'gradebook.BulkManagementTab.successDialog',
    defaultMessage: 'CSV processing. File uploads may take several minutes to complete.',
    description: 'Success Dialog message in BulkManagement Tab File Upload Form',
  },
});

export default messages;
