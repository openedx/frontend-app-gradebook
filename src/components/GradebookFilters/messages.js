import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignments: {
    id: 'gradebook.GradebookFilters.assignmentsFilterLabel',
    defaultMessage: 'Assignments',
    description: 'Assignment filter group label in Gradebook Filters',
  },
  overallGrade: {
    id: 'gradebook.GradebookFilters.overallGradeFilterLabel',
    defaultMessage: 'Overall Grade',
    description: 'Overall Grade filter group label in Gradebook Filters',
  },
  studentGroups: {
    id: 'gradebook.GradebookFilters.studentGroupsFilterLabel',
    defaultMessage: 'Student Groups',
    description: 'Student Groups filter group label in Gradebook Filters',
  },
  includeCourseTeamMembers: {
    id: 'gradebook.GradebookFilters.includeCourseTeamMembersFilterLabel',
    defaultMessage: 'Include Course Team Members',
    description: 'Include Course Team Members filter label in Gradebook Filters',
  },
  assignment: {
    id: 'gradebook.GradebookFilters.assignmentFilterLabel',
    defaultMessage: 'Assignment',
    description: 'Assignment filter select label in Gradebook Filters',
  },
  assignmentTypes: {
    id: 'gradebook.GradebookFilters.assignmentTypesLabel',
    defaultMessage: 'Assignment Types',
    description: 'Assignment Types filter select label in Gradebook Filters',
  },
  maxGrade: {
    id: 'gradebook.GradebookFilters.maxGradeFilterLabel',
    defaultMessage: 'Max Grade',
    description: 'Max-grade filter select label in Gradebook Filters',
  },
  minGrade: {
    id: 'gradebook.GradebookFilters.minGradeFilterLabel',
    defaultMessage: 'Min Grade',
    description: 'Min-grade filter select label in Gradebook Filters',
  },
  cohorts: {
    id: 'gradebook.GradebookFilters.cohorts',
    defaultMessage: 'Cohorts',
    description: 'Cohorts filter select label in Gradebook Filters',
  },
  cohortAll: {
    id: 'gradebook.GradebookFilters.cohortsAll',
    defaultMessage: 'Cohort-All',
    description: 'Cohorts filter select default in Gradebook Filters',
  },
  tracks: {
    id: 'gradebook.GradebookFilters.tracks',
    defaultMessage: 'Tracks',
    description: 'Tracks filter select label in Gradebook Filters',
  },
  trackAll: {
    id: 'gradebook.GradebookFilters.trackAll',
    defaultMessage: 'Track-All',
    description: 'Tracks filter select default in Gradebook Filters',
  },
  closeFilters: {
    id: 'gradebook.GradebookFilters.closeFilters',
    defaultMessage: 'Close Filters',
    description: 'Button label for Close button in Gradebook Filters',
  },
});

export default messages;
