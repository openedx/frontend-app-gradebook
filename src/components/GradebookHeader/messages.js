import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  backToDashboard: {
    id: 'gradebook.GradebookHeader.backButton',
    defaultMessage: 'Back to Dashboard',
    description: 'Button text to take user back to LMS dashboard in Gradebook Header',
  },
  gradebook: {
    id: 'gradebook.GradebookHeader.appLabel',
    defaultMessage: 'Gradebook',
    description: 'Top-level app title in Gradebook Header component',
  },
  frozenWarning: {
    id: 'gradebook.GradebookHeader.frozenWarning',
    defaultMessage: 'The grades for this course are now frozen. Editing of grades is no longer allowed.',
    description: 'Warning message in Gradebook Header for frozen messages',
  },
  unauthorizedWarning: {
    id: 'gradebook.GradebookHeader.unauthorizedWarning',
    defaultMessage: 'You are not authorized to view the gradebook for this course.',
    description: 'Warning message in Gradebook Header when user is not allowed to view the app',
  },
  toActivityLog: {
    id: 'gradebook.GradebookHeader.toActivityLogButton',
    defaultMessage: 'View Bulk Management History',
    description: 'Button text for button navigating to Bulk Managment Activity Log',
  },
  toGradesView: {
    id: 'gradebook.GradebookHeader.toGradesView',
    defaultMessage: 'Return to Gradebook',
    description: 'Button text for button navigating to Grades view.',
  },
});

export default messages;
