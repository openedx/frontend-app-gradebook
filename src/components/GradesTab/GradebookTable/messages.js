import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  emailHeading: {
    id: 'gradebook.GradesTab.table.headings.email',
    defaultMessage: 'Email',
    description: 'Gradebook table email column header',
  },
  totalGradeHeading: {
    id: 'gradebook.GradesTab.table.headings.totalGrade',
    defaultMessage: 'Total Grade (%)',
    description: 'Gradebook table total grade column header',
  },
  usernameHeading: {
    id: 'gradebook.GradesTab.table.headings.username',
    defaultMessage: 'Username',
    description: 'Gradebook table username column header',
  },
  studentKeyLabel: {
    id: 'gradebook.GradesTab.table.labels.studentKey',
    defaultMessage: 'Student Key*',
    description: 'Gradebook table Student Key label',
  },
  usernameLabel: {
    id: 'gradebook.GradesTab.table.labels.username',
    defaultMessage: 'Username',
    description: 'Gradebook table username label',
  },
  totalGradePercentage: {
    id: 'gradebook.GradesTab.table.totalGradePercentage',
    defaultMessage: 'Total Grade values are always displayed as a percentage',
    description: 'Gradebook table message that total grades are displayed in percent format',
  },
});

export default messages;
