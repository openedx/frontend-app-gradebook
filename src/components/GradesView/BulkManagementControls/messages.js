import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  downloadGradesBtn: {
    id: 'gradebook.GradesView.BulkManagementControls.bulkManagementLabel',
    defaultMessage: 'Download Grades',
    description: 'A labeled button that allows an admin user to download course grades all at once (in bulk).',
  },
});

export default messages;
