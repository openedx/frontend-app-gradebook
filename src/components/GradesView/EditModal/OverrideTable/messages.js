import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  adjustedGradeHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.adjustedGradeHeader',
    defaultMessage: 'Adjusted grade',
    description: 'Edit Modal Override Table Adjusted grade column header',
  },
  dateHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.dateHeader',
    defaultMessage: 'Date',
    description: 'Edit Modal Override Table Date column header',
  },
  graderHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.graderHeader',
    defaultMessage: 'Grader',
    description: 'Edit Modal Override Table Grader column header',
  },
  reasonHeader: {
    id: 'gradebook.GradesView.EditModal.Overrides.reasonHeader',
    defaultMessage: 'Reason',
    description: 'Edit Modal Override Table Reason column header',
  },
});

export default messages;
