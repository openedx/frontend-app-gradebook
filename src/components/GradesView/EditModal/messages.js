import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignmentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.assignment',
    defaultMessage: 'Assignment',
    description: 'Edit Modal Assignment header',
  },
  currentGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.currentGrade',
    defaultMessage: 'Current Grade',
    description: 'Edit Modal Current Grade header',
  },
  originalGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.originalGrade',
    defaultMessage: 'Original Grade',
    description: 'Edit Modal Original Grade header',
  },
  studentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.student',
    defaultMessage: 'Student',
    description: 'Edit Modal Student header',
  },
  title: {
    id: 'gradebook.GradesView.EditModal.title',
    defaultMessage: 'Edit Grades',
    description: 'Edit Modal title',
  },
  closeText: {
    id: 'gradebook.GradesView.EditModal.closeText',
    defaultMessage: 'Cancel',
    description: 'Edit Modal close button text',
  },
  visibility: {
    id: 'gradebook.GradesView.EditModal.contactSupport',
    defaultMessage: 'Showing most recent actions (max 5). To see more, please contact support',
    description: 'Edit Modal visibility hint message',
  },
  saveVisibility: {
    id: 'gradebook.GradesView.EditModal.saveVisibility',
    defaultMessage: 'Note: Once you save, your changes will be visible to students.',
    description: 'Edit Modal saved changes effect hint',
  },
  saveGrade: {
    id: 'gradebook.GradesView.EditModal.saveGrade',
    defaultMessage: 'Save Grades',
    description: 'Edit Modal Save button label',
  },
});

export default messages;
