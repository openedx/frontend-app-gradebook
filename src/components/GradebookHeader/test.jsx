import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import actions from 'data/actions';
import selectors from 'data/selectors';
import { views } from 'data/constants/app';
import messages from './messages';
import { GradebookHeader, mapDispatchToProps, mapStateToProps } from '.';

jest.mock('@edx/paragon', () => ({
  Button: () => 'Button',
}));
jest.mock('@edx/frontend-platform/i18n', () => ({
  defineMessages: m => m,
  FormattedMessage: () => 'FormattedMessage',
}));
jest.mock('data/actions', () => ({
  __esModule: true,
  default: {
    app: { setView: jest.fn() },
  },
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    app: {
      activeView: jest.fn(state => ({ aciveView: state })),
      courseId: jest.fn(state => ({ courseId: state })),
    },
    assignmentTypes: { areGradesFrozen: jest.fn(state => ({ areGradesFrozen: state })) },
    roles: { canUserViewGradebook: jest.fn(state => ({ canUserViewGradebook: state })) },
    root: { showBulkManagement: jest.fn(state => ({ showBulkManagement: state })) },
  },
}));

const courseId = 'fakeID';
describe('GradebookHeader component', () => {
  const props = {
    activeView: views.grades,
    areGradesFrozen: false,
    canUserViewGradebook: false,
    courseId,
    showBulkManagement: false,
  };
  beforeEach(() => {
    props.setView = jest.fn();
  });
  describe('snapshots', () => {
    let el;
    beforeEach(() => {
      el = shallow(<GradebookHeader {...props} />);
      el.instance().handleToggleViewClick = jest.fn().mockName('this.handleToggleViewClick');
    });
    describe('default values (grades frozen, cannot view).', () => {
      test('unauthorized warning, but no grades frozen warning', () => {
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
    describe('grades frozen, cannot view', () => {
      test('unauthorized warning, and grades frozen warning.', () => {
        el.setProps({ areGradesFrozen: true });
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
    describe('grades frozen, can view.', () => {
      test('grades frozen warning but no unauthorized warning', () => {
        el.setProps({ areGradesFrozen: true, canUserViewGradebook: true });
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
    describe('show bulk management, active view is grades view', () => {
      test('toggle view button to activity log', () => {
        el.setProps({ showBulkManagement: true });
        expect(el.find(Button).getElement()).toEqual((
          <Button
            variant="tertiary"
            onClick={el.instance().handleToggleViewClick}
          >
            <FormattedMessage {...messages.toActivityLog} />
          </Button>
        ));
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
    describe('show bulk management, active view is bulkManagementHistory view', () => {
      test('toggle view button to grades', () => {
        el.setProps({ showBulkManagement: true, activeView: views.bulkManagementHistory });
        expect(el.find(Button).getElement()).toEqual((
          <Button
            variant="tertiary"
            onClick={el.instance().handleToggleViewClick}
          >
            <FormattedMessage {...messages.toGradesView} />
          </Button>
        ));
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
  });
  describe('behavior', () => {
    let el;
    beforeEach(() => {
      el = shallow(<GradebookHeader {...props} />);
    });
    describe('handleToggleViewClick', () => {
      test('calls setView with activity view if activeView is grades', () => {
        el.instance().handleToggleViewClick();
        expect(props.setView).toHaveBeenCalledWith(views.bulkManagementHistory);
      });
      test('calls setView with grades view if activeView is bulkManagementHistory', () => {
        el.setProps({ activeView: views.bulkManagementHistory });
        el.instance().handleToggleViewClick();
        expect(props.setView).toHaveBeenCalledWith(views.grades);
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { a: 'test', example: 'state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('activeView from app.activeView', () => {
      expect(mapped.activeView).toEqual(selectors.app.activeView(testState));
    });
    test('courseId from app.courseId', () => {
      expect(mapped.courseId).toEqual(selectors.app.courseId(testState));
    });
    test('areGradesFrozen from assignmentTypes selector', () => {
      expect(
        mapped.areGradesFrozen,
      ).toEqual(selectors.assignmentTypes.areGradesFrozen(testState));
    });
    test('canUserViewGradebook from roles selector', () => {
      expect(
        mapped.canUserViewGradebook,
      ).toEqual(selectors.roles.canUserViewGradebook(testState));
    });
    test('showBulkManagement from root showBulkManagement selector', () => {
      expect(mapped.showBulkManagement).toEqual(selectors.root.showBulkManagement(testState));
    });
  });

  describe('mapDispatchToProps', () => {
    test('setView from actions.app.setView', () => {
      expect(mapDispatchToProps.setView).toEqual(actions.app.setView);
    });
  });
});
