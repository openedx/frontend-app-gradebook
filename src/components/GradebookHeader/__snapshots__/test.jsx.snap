// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradebookHeader component snapshots default values (grades frozen, cannot view). unauthorized warning, but no grades frozen warning 1`] = `
<div
  className="gradebook-header"
>
  <a
    className="mb-3"
    href="http://localhost:18000/courses/fakeID/instructor"
  >
    <span
      aria-hidden="true"
    >
      &lt;&lt; 
    </span>
    <FormattedMessage
      defaultMessage="Back to Dashboard"
      description="Button text to take user back to LMS dashboard in Gradebook Header"
      id="gradebook.GradebookHeader.backButton"
    />
  </a>
  <h1>
    <FormattedMessage
      defaultMessage="Gradebook"
      description="Top-level app title in Gradebook Header component"
      id="gradebook.GradebookHeader.appLabel"
    />
  </h1>
  <div
    className="subtitle-row d-flex justify-content-between align-items-center"
  >
    <h3>
      fakeID
    </h3>
  </div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="You are not authorized to view the gradebook for this course."
      description="Warning message in Gradebook Header when user is not allowed to view the app"
      id="gradebook.GradebookHeader.unauthorizedWarning"
    />
  </div>
</div>
`;

exports[`GradebookHeader component snapshots grades frozen, can view. grades frozen warning but no unauthorized warning 1`] = `
<div
  className="gradebook-header"
>
  <a
    className="mb-3"
    href="http://localhost:18000/courses/fakeID/instructor"
  >
    <span
      aria-hidden="true"
    >
      &lt;&lt; 
    </span>
    <FormattedMessage
      defaultMessage="Back to Dashboard"
      description="Button text to take user back to LMS dashboard in Gradebook Header"
      id="gradebook.GradebookHeader.backButton"
    />
  </a>
  <h1>
    <FormattedMessage
      defaultMessage="Gradebook"
      description="Top-level app title in Gradebook Header component"
      id="gradebook.GradebookHeader.appLabel"
    />
  </h1>
  <div
    className="subtitle-row d-flex justify-content-between align-items-center"
  >
    <h3>
      fakeID
    </h3>
  </div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="The grades for this course are now frozen. Editing of grades is no longer allowed."
      description="Warning message in Gradebook Header for frozen messages"
      id="gradebook.GradebookHeader.frozenWarning"
    />
  </div>
</div>
`;

exports[`GradebookHeader component snapshots grades frozen, cannot view unauthorized warning, and grades frozen warning. 1`] = `
<div
  className="gradebook-header"
>
  <a
    className="mb-3"
    href="http://localhost:18000/courses/fakeID/instructor"
  >
    <span
      aria-hidden="true"
    >
      &lt;&lt; 
    </span>
    <FormattedMessage
      defaultMessage="Back to Dashboard"
      description="Button text to take user back to LMS dashboard in Gradebook Header"
      id="gradebook.GradebookHeader.backButton"
    />
  </a>
  <h1>
    <FormattedMessage
      defaultMessage="Gradebook"
      description="Top-level app title in Gradebook Header component"
      id="gradebook.GradebookHeader.appLabel"
    />
  </h1>
  <div
    className="subtitle-row d-flex justify-content-between align-items-center"
  >
    <h3>
      fakeID
    </h3>
  </div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="The grades for this course are now frozen. Editing of grades is no longer allowed."
      description="Warning message in Gradebook Header for frozen messages"
      id="gradebook.GradebookHeader.frozenWarning"
    />
  </div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="You are not authorized to view the gradebook for this course."
      description="Warning message in Gradebook Header when user is not allowed to view the app"
      id="gradebook.GradebookHeader.unauthorizedWarning"
    />
  </div>
</div>
`;

exports[`GradebookHeader component snapshots show bulk management, active view is bulkManagementHistory view toggle view button to grades 1`] = `
<div
  className="gradebook-header"
>
  <a
    className="mb-3"
    href="http://localhost:18000/courses/fakeID/instructor"
  >
    <span
      aria-hidden="true"
    >
      &lt;&lt; 
    </span>
    <FormattedMessage
      defaultMessage="Back to Dashboard"
      description="Button text to take user back to LMS dashboard in Gradebook Header"
      id="gradebook.GradebookHeader.backButton"
    />
  </a>
  <h1>
    <FormattedMessage
      defaultMessage="Gradebook"
      description="Top-level app title in Gradebook Header component"
      id="gradebook.GradebookHeader.appLabel"
    />
  </h1>
  <div
    className="subtitle-row d-flex justify-content-between align-items-center"
  >
    <h3>
      fakeID
    </h3>
    <Button
      onClick={[MockFunction this.handleToggleViewClick]}
      variant="tertiary"
    >
      <FormattedMessage
        defaultMessage="Return to Gradebook"
        description="Button text for button navigating to Grades view."
        id="gradebook.GradebookHeader.toGradesView"
      />
    </Button>
  </div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="You are not authorized to view the gradebook for this course."
      description="Warning message in Gradebook Header when user is not allowed to view the app"
      id="gradebook.GradebookHeader.unauthorizedWarning"
    />
  </div>
</div>
`;

exports[`GradebookHeader component snapshots show bulk management, active view is grades view toggle view button to activity log 1`] = `
<div
  className="gradebook-header"
>
  <a
    className="mb-3"
    href="http://localhost:18000/courses/fakeID/instructor"
  >
    <span
      aria-hidden="true"
    >
      &lt;&lt; 
    </span>
    <FormattedMessage
      defaultMessage="Back to Dashboard"
      description="Button text to take user back to LMS dashboard in Gradebook Header"
      id="gradebook.GradebookHeader.backButton"
    />
  </a>
  <h1>
    <FormattedMessage
      defaultMessage="Gradebook"
      description="Top-level app title in Gradebook Header component"
      id="gradebook.GradebookHeader.appLabel"
    />
  </h1>
  <div
    className="subtitle-row d-flex justify-content-between align-items-center"
  >
    <h3>
      fakeID
    </h3>
    <Button
      onClick={[MockFunction this.handleToggleViewClick]}
      variant="tertiary"
    >
      <FormattedMessage
        defaultMessage="View Bulk Management History"
        description="Button text for button navigating to Bulk Managment Activity Log"
        id="gradebook.GradebookHeader.toActivityLogButton"
      />
    </Button>
  </div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <FormattedMessage
      defaultMessage="You are not authorized to view the gradebook for this course."
      description="Warning message in Gradebook Header when user is not allowed to view the app"
      id="gradebook.GradebookHeader.unauthorizedWarning"
    />
  </div>
</div>
`;
